lib/LWP/UserAgent.pm

```diff
@@ -1039,6 +1043,11 @@ sub mirror
             rename( $tmpfile, $file )
                 or die "Cannot rename '$tmpfile' to '$file': $!\n";
 
+            my $umask = umask();
+            my $mode = 0666 &~ ($umask || 0);
```

Comment:
I'm undecided if `0666` is good when `umask` is not supported (== returns `undef`), or if we should use some other perms (`0644`?) or skip `chmod` altogether when that's the case.